cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -286,10 +321,78 @@ public ConfigMap generateMetricsConfigMap() {
         if (isMetricsEnabled) {
             volumeMountList.add(createVolumeMount(metricsConfigVolumeName, metricsConfigMountPath));
         }
+        if (rackConfig != null) {
+            volumeMountList.add(createVolumeMount(KAFKA_INIT_VOLUME_NAME, KAFKA_INIT_VOLUME_MOUNT));
+        }
 
         return volumeMountList;
     }
 
+    @Override
+    protected Affinity getAffinity() {
+
+        List<WeightedPodAffinityTerm> weightedPodAffinityTerms = new ArrayList<>();
+        PodAntiAffinity podAntiAffinity = null;
+        Affinity affinity = null;
+
+        // adding the affinity term for rack feature only if it's enabled
+        if (rackConfig != null) {
+            Map<String, String> matchLabels = new HashMap<>();
+            matchLabels.put(Labels.STRIMZI_CLUSTER_LABEL, cluster);
+            matchLabels.put(Labels.STRIMZI_NAME_LABEL, name);
+            matchLabels.put(Labels.STRIMZI_TYPE_LABEL, AssemblyType.KAFKA.toString());
+            LabelSelector labelSelector = new LabelSelector(null, matchLabels);
+
+            PodAffinityTerm podAffinityTerm = new PodAffinityTerm(labelSelector, null, rackConfig.getTopologyKey());
+            WeightedPodAffinityTerm weightedPodAffinityTerm = new WeightedPodAffinityTerm(podAffinityTerm, 100);
+            weightedPodAffinityTerms.add(weightedPodAffinityTerm);
+        }
+
+        if (weightedPodAffinityTerms.size() > 0) {
+            podAntiAffinity = new PodAntiAffinityBuilder()
+                    .withPreferredDuringSchedulingIgnoredDuringExecution(weightedPodAffinityTerms)
+                    .build();
+        }
+
+        if (podAntiAffinity != null) {
```

Comment:
I don't understand why you've coded this as 3 separate `if` statements, given that each sets the condition that the following `if` statement will test. It makes the code harder to understand imho, is there a reason for it?

Reply:
Well my intention was having this `getAffinity` method used for the future support for more "customized" pod/node affinity. The rack pod anti affinity is just the first. The final intention is also returning `null` if no affinity are defined (avoiding an empty `affinity` list created by Kubernetes/OpenShift). I could review the code of course.